aea/skills/base/core.py

```diff
@@ -101,6 +173,7 @@ def parse_module(cls, path: str, behaviours_configs: List[BehaviourConfig]) -> L
             else:
                 args = behaviour_config.args
                 behaviour = behaviour_class(**args)
+                behaviour._context = skill_context
```

Comment:
If we parse it to the initialiser then the inherited classes need to be aware of it..not ideal.